src/web/components/StickyBottomBanner/StickyBottomBanner.tsx

```diff
@@ -13,53 +14,52 @@ type Props = {
 };
 
 const getEngagementBannerLastClosedAt = (): string | undefined => {
-    return localStorage.getItem('gu.prefs.engagementBannerLastClosedAt') || undefined;
+    return (
+        localStorage.getItem('gu.prefs.engagementBannerLastClosedAt') ||
+        undefined
+    );
 };
 
 export const StickyBottomBanner = ({
     isSignedIn,
     countryCode,
     CAPI,
 }: Props) => {
-    const [showCMP, setShowCMP] = useState<boolean | null>(null);
+    const [showOldCMP, setShowOldCMP] = useState<boolean | null>(null);
+    const [newCMPWillShow, setNewCMPWillShow] = useState<boolean | null>(null);
 
     useEffect(() => {
-        const callShouldShow = () => setShowCMP(shouldShowCMP());
-
-        if (CAPI.config.cmpUi) {
-            callShouldShow();
-        }
+        Promise.all([shouldShowOldCMP(), willShowNewCMP()]).then(
+            ([shouldShowOld, willShowNew]) => {
+                setShowOldCMP(shouldShowOld && CAPI.config.cmpUi);
+                setNewCMPWillShow(willShowNew);
+            },
+        );
     }, [CAPI.config.cmpUi]);
 
-    // Don't render anything until we know whether we can show the CMP
-    if (showCMP === null) {
-        return null;
-    }
+    if (newCMPWillShow) return null;
+
+    if (showOldCMP) return <CMP />;
```

Comment:
Do we need two pieces of state here, as you wait for the promise.all to resolve with both values so you can have a single pieces of state with an if and if else statement?

Reply:
i had to rejig the logic, sorry

Reply:
not sure i get you - `newCMPWillShow` could be false and `showOldCMP`, what would the single piece of state look like?